pkg/controller/k8s/network_policy_controller_test.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2021 The Lynx Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	securityv1alpha1 "github.com/smartxworks/lynx/pkg/apis/security/v1alpha1"
+)
+
+var _ = Describe("pod controller", func() {
+	var ctx context.Context
+
+	BeforeEach(func() {
+		ctx = context.Background()
+	})
+
+	Context("Test network policy sync to security policy", func() {
+		protoTCP := corev1.ProtocolTCP
+		port80 := intstr.FromInt(80)
+		// port8000 := intstr.FromInt(80)
+		networkPolicy := &networkingv1.NetworkPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "network-policy",
+				Namespace: "default",
+			},
+			Spec: networkingv1.NetworkPolicySpec{
+				PodSelector: metav1.LabelSelector{
+					MatchLabels: map[string]string{"label": "value"},
+				},
+				Ingress: []networkingv1.NetworkPolicyIngressRule{
+					{
+						Ports: []networkingv1.NetworkPolicyPort{
+							{
+								Protocol: &protoTCP,
+								Port:     &port80,
+								// EndPort:  &port8000.IntVal,
+							},
+						},
+						From: []networkingv1.NetworkPolicyPeer{},
+					},
+				},
+				PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
```

Comment:
Add some more test cover the egress, udp protocol, icmp protocol etc.

Reply:
More testcases has been added.
This part will only generate securityPolicy, so UT only needs to check the existence of this value in SecurityPolicy.
Something wrong with codecov, the coverage of rule.go varies widely ( from 50% to 90%)